### DISCUSSION TITLE
What have you written that uses Azure Mobile Apps?

### DISCUSSION THREAD

**Discussion by adrianhall** [Azure/azure-mobile-apps#647] [Show and tell]
I tend to only hear about issues y'all are concerned about, which helps push the product forward.  But I'd like to hear about what things you are building.

So, what are you building?

* What platforms are you targeting?
* What are you writing the app in?
* Is it published?
* What does offline support mean to you?
* Can you share a screen shot or two of your work?

Brag about what you are using this stuff for!  Feel free to let me know what worked well and what didn't as well (even if it isn't a bug).

**Comment by Arslan007** [comment]
Hi @adrianhall 

its a great product and we are loving every bit of it. 

1. What platforms are you targeting?
- MAUI
2. What are you writing the app in?
- Our future app is for assigning the tasks and maintaining the work history etc..
3.  is it published?
- We are still cooking it in MAUI and it will be in house enterprise app. Earlier we have had our apps in Xamarin and our own sync mechanism
4.  What does offline support mean to you?
- Offline support means alot and its a great feature. Specially needed when you are in field and have no internet connectivity and you are still able to work with this and when you are online again then all works like charm 
5.  Can you share a screen shot or two of your work?
- Hopefully yes in near future :)

**Reply by adrianhall** [reply]
Love the MAUI work!  Can't wait for the screen shots!

**Comment by amertlich** [comment]
We appreciate your work keeping this project going and use it mainly for the offline sync capabilities.

Currently we have WPF and Xamarin.Forms Android app released with 4.2 and are in the process of updating to v5. Our next gen versions will be targeting .NET 6 and Standard 2.1 (Still WPF/Android, and initially still Xamarin), with MAUI (iOS + Android) and Blazor (Web\Desktop) additions hopefully coming very soon.  We are slow moving, working on other projects/products simultaneously, so we've appreciated the extended support for Xamarin.

The offline features are the main reason we adopted this library. We essentially distribute the offline database as its own file format. It's a local-first design, so cloud is completely optional but highly valued for multi-user collaboration. You can find our current Android version [here](https://play.google.com/store/apps/details?id=com.junipersys.uinta) although big improvements are coming soon.

Our next gen version makes heavy use of Sqlite's JSON1 (and Postgres' JSONB in backend) features (as we transition away from a traditional EAV model). Out of the box, nested JSON queries aren't compatible with the current v5 library so we have to roll our own queries in many instances.  For the most part the Linq->ODATA features are very valuable, especially for our future web based Blazor client and its ability to query the remote tables the same way we query offline tables in our other clients.  But the JSON queries have to be custom handled external to this library. We don't expect this to be supported by the library (although if you could figure that out it would be awesome). Just making you aware that the Datasync SDK isn't the only SQL connection instance we are maintaining to our databases, but the datasync.client connection is the one that's handling all writes to the database (that we care to synchronize) so that the operations queue is appropriately tracked.

Another unique aspect is we do enforce foreign keys both client and server side. This has worked well despite the recommendations to avoid these sorts of dependencies between tables, but has introduced some complexities to make sure things are synchronized in correct orders and conflicts are appropriately handled.

Our biggest complaint is how slow push operations are in large batches, which seems like something you may be improving soon.  Richer string comparison queries would also be nice (i.e. contains with case insensitivity, or regex support) but not a must have feature.

Overall we are big fans of the library and value the work you've done!  Thanks!

**Comment by ideas2appsaustralia** [comment]
Adrian I've built a .net maui app for a local plumbing business to replace a quality control step they need to do in the field which was previously done on paper. With an azure sql dbase on the backend they can work offline on site recording their test results then upload whenever they have internet again. The app can also print to pdf reports for their clients. They have to test to certain specifications so the app can indicate whenever their entered results are out of range of specification. Functionality like this can only come with an app. It removes retyping back at office, all data is now in a dbase so they can retrieve quickly and do things with it and a quicker turn around time for clients and a stronger compliance capability.

what I am requiring next is a blazer web app so they can perform administration on the data, alter data, download and access in excel and upload the lists for each site so they have the number of test points preloaded for each job.

I can't recall what you are going to do with blazer since your earlier comments on it but Im assuming ability to use data sync client with blazer will eventually come still? the blazor I will build though for above client will just the usual API to work with dbase. However I have another job which would be a blazer PWA and used in the field so this capability with data sync will be very useful as it will provide another platform for clients who may not want a mobile app

thanks heaps for building this neat capability. Offline capability is a must and enables so many opportunities for businesses that are working in the field.

Scott
